Hearthstone Deck Tracker/Core.cs

```diff
@@ -169,6 +175,42 @@ public static async void Initialize()
 			);
 		}
 
+		private static void CheckForHearthstoneUpdate()
+		{
+			try
+			{
+				var hearthDbVersion = HearthDb.Info.HearthDbVersion.ToString();
+				if(hearthDbVersion != Config.Instance.HearthdbVersion)
+				{
+					AssetDownloaders.cardImageDownloader.ClearStorage();
+					Config.Instance.HearthdbVersion = hearthDbVersion;
+					Config.Save();
+				}
+			}
+			catch(Exception ex)
+			{
+				Log.Error($"Could not check for updated HearthDB version: {ex}");
+			}
+		}
+
+		private static void CheckForCardImageUpdate()
```

Comment:
You can simplify all of this a little bit by doing this:
```diff
index c1cc788c..ec3343aa 100644
--- a/Hearthstone Deck Tracker/Core.cs	
+++ b/Hearthstone Deck Tracker/Core.cs	
@@ -152,7 +152,7 @@ namespace Hearthstone_Deck_Tracker
 			}
 			LogWatcherManger.Start(Game).Forget();
 
-			RemoteConfig.Instance.Loaded += (ConfigData data) => CheckForCardImageUpdate();
+			RemoteConfig.Instance.Loaded += CheckForCardImageUpdate;
 			RemoteConfig.Instance.Load();
 			HotKeyManager.Load();
 
@@ -193,15 +193,16 @@ namespace Hearthstone_Deck_Tracker
 			}
 		}
 
-		private static void CheckForCardImageUpdate()
+		private static void CheckForCardImageUpdate(ConfigData rConfig)
 		{
 			try
 			{
 				var hearthDbVersion = HearthDb.Info.HearthDbVersion.ToString();
-				if(RemoteConfig.Instance?.Data?.UpdateInfo?.Version != Config.Instance.RemoteHearthstoneVersion)
+				var remoteVersion = rConfig?.UpdateInfo?.Version;
+				if(remoteVersion.HasValue && remoteVersion > Config.Instance.RemoteHearthstoneVersion)
 				{
 					AssetDownloaders.cardImageDownloader.ClearStorage();
-					Config.Instance.RemoteHearthstoneVersion = RemoteConfig.Instance?.Data?.UpdateInfo?.Version ?? 0;
+					Config.Instance.RemoteHearthstoneVersion = remoteVersion.Value;
 					Config.Save();
 				}
 			}
```

- Loaded is called with the config data already so you don't need to manually look it up again
- We should explicitly check that ` remoteVersion > localVersion` rather than `remoteVersion != localVersion`